File: Set 6/Form the Largest Number/LargestNumber.py
def largest_number(arr):
    arr = list(map(str, arr))
    sorted_arr = sorted(arr, key=lambda x: x * 10, reverse=True)
    result = ''.join(sorted_arr)
    return '0' if result[0] == '0' else result


t = int(input())
for _ in range(t):
    arr = list(map(int, input().split()))
    print(largest_number(arr))

File: Set 6/Form the Largest Number/LargestNumber.cpp
#include <iostream>
#include <vector>
#include <string>
#include <algorithm>
using namespace std;

bool comparer(const string &x, const string &y) {
    return x + y > y + x;
}

string largestNumber(vector<int> &arr) {
    vector<string> strArr;
    for (int num : arr) {
        strArr.push_back(to_string(num));
    }

    sort(strArr.begin(), strArr.end(), comparer);

    string result;
    for (const string &s : strArr) {
        result += s;
    }

    if (result[0] == '0') {
        return "0";
    }

    return result;
}

int main() {
    int t;
    cin >> t;
    while (t--) {
        int n;
        cin >> n;
        vector<int> arr(n);
        for (int i = 0; i < n; ++i) {
            cin >> arr[i];
        }
        cout << largestNumber(arr) << endl;
    }
    return 0;
}
